### PR TITLE
adding the new SpectrogramSequence

### PR DESCRIPTION
**Pull request by abit2**
initial commit now working :)
@DanRyanIrish

### CODE REVIEW

**Comment by abit2** [comment]
https://github.com/sunpy/irispy/issues/47

File: irispy/spectrograph.py [hunk 1]
```diff
@@ -151,3 +156,85 @@ def _enter_column_into_table_as_quantity(header_property_name, header, header_co
...
+            # appending all the indexes to be returned in the answer
+                indexed_not_as_one.append(wcs.naxis-1-i)
+        list_arguemnts = list_arg[::-1]
+        pixel_to_world = wcs.all_pix2world(*list_arguemnts, origin)
+        # collecting all the needed answer in this list.
+        for index in indexed_not_as_one[::-1]:
+            result.append(u.Quantity(pixel_to_world[index], unit=wcs.wcs.cunit[index]))
+        return result[::-1]
+
+class _IndexAsRasterSlicer(object):
```
Comment (DanRyanIrish) [inline irispy/spectrograph.py]: Call this ```_IndexByRasterSilcer```

File: irispy/spectrograph.py [hunk 2]
```diff
@@ -151,3 +156,85 @@ def _enter_column_into_table_as_quantity(header_property_name, header, header_co
...
+            raster_each_scan.append(scan.dimensions.shape[0].value)
+        return raster_each_scan
+
+    @property
+    def first_exposure_raster_position(self):
+        return "Raster Position of first exposure : " + str(self.data[0].dimensions.shape[0].value)
+
+    @property
+    def index_by_raster(self):
+        return _IndexAsRasterSlicer(self)
```
Comment (DanRyanIrish) [inline irispy/spectrograph.py]: See comment below about the name of this function.

File: irispy/spectrograph.py [hunk 3]
```diff
@@ -151,3 +156,85 @@ def _enter_column_into_table_as_quantity(header_property_name, header, header_co
         raise ValueError("Multiple property names equal to {0}".format(header_property_name))
     pop_colname = header_colnames.pop(index)
     return Quantity(data[:, header[pop_colname]], unit=unit)
+
+
+class SpectrogramSequence(NDCubeSequence):
+    """docstring for SpectrogramSequence"""
+
+    def __init__(self, data_list, meta=None, common_axis=None, **kwargs):
```
Comment (DanRyanIrish) [inline irispy/spectrograph.py]: For this object, ```common_axis``` should be a required arg, not a kwarg.  Also, I think ```raster_positions_per_scan``` and ```first_exposure_raster_position``` should be user args, as the shape of the underlying ```NDCubeSequence``` may not reflect the rastering mode itself.

File: irispy/spectrograph.py [hunk 4]
```diff
@@ -151,3 +156,85 @@ def _enter_column_into_table_as_quantity(header_property_name, header, header_co
...
+            data_list, meta=meta, common_axis=common_axis, **kwargs)
+
+    def __getitem__(self, item):
+        if item is None or (isinstance(item, tuple) and None in item):
+            raise IndexError("None indices not supported")
+        # need to slice the time here
+        return self.index_as_cube[item]
+
+    @property
+    def raster_position_per_scan(self):
```
Comment (DanRyanIrish) [inline irispy/spectrograph.py]: See comment on L164

File: irispy/spectrograph.py [hunk 5]
```diff
@@ -151,3 +156,85 @@ def _enter_column_into_table_as_quantity(header_property_name, header, header_co
...
+        raster_each_scan = []
+        for scan in self.data:
+            raster_each_scan.append(scan.dimensions.shape[0].value)
+        return raster_each_scan
+
+    @property
+    def first_exposure_raster_position(self):
```
Comment (DanRyanIrish) [inline irispy/spectrograph.py]: See comment on L164

**Comment by abit2** [comment]
@DanRyanIrish

**Pull request merged**